tests/extmod/vfs_fat_ramdisk.py

```diff
@@ -0,0 +1,48 @@
+import sys
+import uos
+try:
+    uos.VfsFat
+except AttributeError:
+    print("SKIP")
+    sys.exit()
+
+
+class RAMFS:
+    def __init__(self, blocks):
+        self.data = bytearray(blocks * 512)
+
+    def readblocks(self, n, buf):
+        #print("readblocks(%s, %x(%d))" % (n, id(buf), len(buf)))
+        for i in range(len(buf)):
+            buf[i] = self.data[n*512+i]
+
+    def writeblocks(self, n, buf):
+        #print("writeblocks(%s, %x)" % (n, id(buf)))
+        for i in range(len(buf)):
+            self.data[n*512+i] = buf[i]
+
+    def sync(self):
+        pass
+
+    def count(self):
```

Comment:
This is using old-style protocol.  Might be good to change it to new ioctl style so other people who copy the code also use the new style.